iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/StaticVarCompensatorImpl.java

```diff
@@ -126,15 +122,15 @@ public StaticVarCompensatorImpl setReactivePowerSetpoint(double reactivePowerSet
     @Override
     public RegulationMode getRegulationMode() {
         int variantIndex = getNetwork().getVariantIndex();
-        return regulationMode.get(variantIndex) != -1 ? RegulationMode.values()[regulationMode.get(variantIndex)] : null;
+        return regulatingPoint.getRegulationMode(variantIndex) != -1 ? RegulationMode.values()[regulatingPoint.getRegulationMode(variantIndex)] : null;
     }
 
     @Override
     public StaticVarCompensatorImpl setRegulationMode(RegulationMode regulationMode) {
         NetworkImpl n = getNetwork();
         ValidationUtil.checkSvcRegulator(this, getVoltageSetpoint(), getReactivePowerSetpoint(), regulationMode, n.getMinValidationLevel());
         int variantIndex = n.getVariantIndex();
-        RegulationMode oldValue = RegulationMode.values()[this.regulationMode.set(variantIndex,
+        RegulationMode oldValue = RegulationMode.values()[regulatingPoint.setRegulationMode(variantIndex,
                 regulationMode != null ? regulationMode.ordinal() : -1)];
```

Comment:
if `regulationMode == null` you put `-1` so you shouldn't look into the array without checking `!=-1`